lib/client.js

```diff
@@ -158,6 +158,123 @@ class Client {
     const relativePathWithQuery = `${params.resourceType}?${new URLSearchParams(params.searchParams)}`;
     return this.httpClient.get(relativePathWithQuery);
   }
+
+  /**
+   * Create a resource by FHIR id.
+   *
+   * @example
+   * const newPatient = { resourceType: 'Patient',
+   *                      active: true,
+   *                      name: [{ use: 'official', family: ['Coleman'], given: ['Lisa', 'P.'] }],
+   *                      gender: 'female',
+   *                      birthDate: '1948-04-14',
+   *                    }
+   *
+   * // Using promises
+   * fhirClient.create({ resourceType: 'Patient',
+   *                     resource: newPatient,
+   *                   })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.create({ resourceType: 'Patient',
+   *                                          resource: newPatient,
+   *                                         })
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resource - The new resource data to create.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  create(params) {
+    return this.httpClient.post(params.resourceType, params.resource);
+  }
+
+  /**
+   * Delete a resource by FHIR id.
+   *
+   * @example
+   *
+   * // Using promises
+   * fhirClient.delete({ resourceType: 'Patient', id: 12345 })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.delete({ resourceType: 'Patient', id: 12345 });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
+   * @param {string} params.id - The FHIR id for the resource.
+   *
+   * @return {Promise<Object>} FHIR resource
+   */
+  delete(params) {
+    return this.httpClient.delete(`${params.resourceType}/${params.id}`);
+  }
+
+  /**
+   * Update a resource by FHIR id.
+   *
+   * @example
+   *
+   * const updatedPatient = { resourceType: 'Patient',
+   *                          birthDate: '1948-04-14' }
+   * // Using promises
+   * fhirClient.update({ resourceType: 'Patient',
+   *                     id: 12345,
+   *                     resource: updatedPatient,
+   *                   })
+   *           .then((data) => { console.log(data); });
+   *
+   * // Using async
+   * let response = await fhirClient.update({ resourceType: 'Patient',
+   *                                          id: 12345,
+   *                                          resource: updatedPatient,
+   *                                        });
+   * console.log(response);
+   *
+   * @param {Object} params - The request parameters.
+   * @param {string} params.resourceType - The resource type (e.g. "Patient", "Observation").
```

Comment:
I wonder if this PR could define a  "resource" as a (reusable?) "Type definition" -- as described, here:

<img width="1400" alt="image" src="https://user-images.githubusercontent.com/3383704/38432491-9e647f74-3995-11e8-9491-c30569d2bcae.png">

http://usejsdoc.org/tags-type.html

Reply:
e.g.:
```
/**
  @typedef FHIRResource
  @type {object}
  @property {string} id - an ID.
  @property {string} resourceType - your name.
  // etc.

/** @type {FHIRResource} /
var props;

```

Reply:
e.g.:
```
/**
  @typedef FHIRResource
  @type {object}
  @property {string} id - an ID.
  @property {string} resourceType - your name.
  // etc.
*/

// Later:
/** @type {FHIRResource} /
   var props;
  // etc.
*/
```

Reply:
I don't understand what that would gain us. The only thing you can say about all (json) FHIR resources is they have to have a `resourceType`.

Reply:
^ the definition, here, says it also has an `id` ?

...It seems the `@typedef` can be referenced multiple times, so using it would obviate repetition of the fact that it's an `Object`, with a `resourceType` and `id` ?

Reply:
Not sure if "FHIR Resource" would be the type, but the jsdoc-definition, ^above, says it also has an `id` ?  (e.g.: is this actually a "JSON body payload" vs. a set of "request parameters" ?)

...It seems the `@typedef` can be referenced multiple times, so using it would obviate repetition of the fact that it's an `Object`, with a `resourceType` and `id` ?

Reply:
`resourceType` and `id` (when present) are used to construct the appropriate url (`base/resourceType` or `base/resourceType/id`). `resource` (when present) is a json payload.

Reply:
FhirParams? ...as a repeating type?

...with `resource` referencing its own @typedef?  (optional, here)

Reply:
FhirParams? ...as a repeating type?

...with `resource` (or: `payload`?)  referencing its own `@typedef`?  (which would be optional, in `FhirParams` ?)